p2p/host/resource-manager/rcmgr.go

```diff
@@ -316,14 +316,27 @@ func (r *resourceManager) nextStreamId() int64 {
 	return r.streamId
 }
 
-// OpenConnectionNoIP is like OpenConnection, but does not use IP information.
-// Used when we still want to limit the connection by other scopes, but don't
-// have IP information like when we are relaying.
-func (r *resourceManager) OpenConnectionNoIP(dir network.Direction, usefd bool, endpoint multiaddr.Multiaddr) (network.ConnManagementScope, error) {
-	return r.openConnection(dir, usefd, endpoint, netip.Addr{})
+func hasIPInMultiaddr(addr multiaddr.Multiaddr) bool {
+	foundIP := false
+	multiaddr.ForEach(addr, func(c multiaddr.Component) bool {
+		switch c.Protocol().Code {
+		case multiaddr.P_IP4, multiaddr.P_IP6:
+			foundIP = true
+			return false
+		default:
+			return true
```

Comment:
Should this be `return false`?

Alternatively, we can also just assume that if `manet.ToIP` errors, it's not an IP address.